analytics/src/main/java/com/segment/analytics/Analytics.java

```diff
@@ -284,14 +285,25 @@ public void run() {
 
     logger.debug("Created analytics client for project with tag:%s.", tag);
 
+    PackageInfo packageInfo = getPackageInfo(application);
+    final String currentVersion = packageInfo.versionName;
+    final int currentBuild = packageInfo.versionCode;
+
     activityLifecycleCallback =
         new Application.ActivityLifecycleCallbacks() {
           final AtomicBoolean trackedApplicationLifecycleEvents = new AtomicBoolean(false);
+          final AtomicInteger numberOfActivities = new AtomicInteger(1);
+          final AtomicBoolean isChangingActivityConfigurations = new AtomicBoolean(false);
+          final AtomicBoolean firstLaunch = new AtomicBoolean(false);
```

Comment:
Android has new application lifecycle callbacks via `androidx.lifecycle`, but we can't use them due to the dependency requirements AndroidX places. Additionally that breaks the React Native wrapper as AndroidX deps won't work with new RN projects out of the box. We use the traditional method of activity reference counting to detect if we are launching a new activity or resuming to foreground.